packages/react/src/pages/analytics-geography/analytics-geography.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { Dashboard } from '../../components/dashboard/Dashboard';
+import { DashboardCardsGroup } from '../../components/dashboard/DashboardCardGroup';
+import { getSalesByStateAndCity, getSalesByState } from 'dx-rwa-data';
+import {
+  ANALYTICS_PERIODS,
+  DEFAULT_ANALYTICS_PERIOD_KEY,
+} from '../../shared/constants';
+import { Item } from 'devextreme-react/toolbar';
+import * as mapsData from 'devextreme/dist/js/vectormap-data/usa.js';
+import { RevenueByStatesCard } from './cards/RevenueByStatesCard';
+import { RevenueAnalysisByStatesCard } from './cards/RevenueAnalysisByStatesCard';
+import { RevenueSnapshotByStatesCard } from './cards/RevenueSnapshotByStates';
+
+const createMapCoords = (coords: string) => coords.split(', ').map(parseFloat);
+
+export const AnalyticsGeography = () => {
+  const [tabIndex, setTabIndex] = useState(
+    ANALYTICS_PERIODS[DEFAULT_ANALYTICS_PERIOD_KEY].index
+  );
+  const [dateRange, setDateRange] = useState(
+    ANALYTICS_PERIODS[DEFAULT_ANALYTICS_PERIOD_KEY].period.split('/')
+  );
+  const [salesByStateAndCity, setSalesByStateAndCity] = useState([]);
+  const [salesByState, setSalesByState] = useState([]);
+  const [salesByStateMarkers, setSalesByStateMarkers] = useState({});
+
+  useEffect(() => {
+    getSalesByStateAndCity(...dateRange).then((data) => {
+      setSalesByStateAndCity(data);
+      const salesByStateResult = getSalesByState(data);
```

Comment:
Can we get rid of the variable and use the salesByState?

Reply:
No. salesByState is a React state and cannot be mutated directly. We need to use the function result twice, so we need a temporary variable here.